File: schemacrawler-distrib/src/examples/resources/examples/ruby/tables.rb
# Copyright (c) Sualeh Fatehi
# SPDX-License-Identifier: EPL-2.0

require 'java'

def schemacrawler
  Java::Schemacrawler
end

puts catalog.crawlInfo

for table in catalog.tables
  puts ''
  puts table.fullName
  for childTable in table.referencingTables
    puts "  [child] " + childTable.fullName
  end
end

File: schemacrawler-distrib/src/examples/resources/examples/python/tables.py
from schemacrawler.schema import TableRelationshipType

print(catalog.crawlInfo)

for table in catalog.tables:
  print('')
  print(table.fullName)
  for childTable in table.getRelatedTables(TableRelationshipType.child):
    print("  [child] " + childTable.fullName)
